### PR TITLE
Add CI via GitHub Actions

### LINKED ISSUE #5: Add CI
Add a CI to:

- [x] (Lint)
- [x] Build & auto deploy

### PR DESCRIPTION
**Pull request by Gum-Joe**
Add a CI script to:

1. Install deps
2. Lint the code
3. Deploy the code (this deployment only occurs on `master`)

NOTE: 1. and 2. run for pushes to `master` and PRs into it,

TODO: Configure ESLINT

Closes #5

### CODE REVIEW

**Comment by davwheat** [comment]
We should transition to Typescript. I'll do that in another branch **after** #7 is merged (as that has some TS already).

**Comment by Gum-Joe** [comment]
Makes sense, and is what I would also recommend

**Comment by davwheat** [comment]
This could have led to very bad results...

https://github.com/davwheat/world-cup-of-tocs/blob/60e70c091fd3e08d94630b80a968791b34c11a5c/.github/workflows/build-deploy.yml#L126

Any push to a branch containing `master` in it would result in a deploy, as well as a PR from `master` to another branch (if we ever wanted to do that for some mad reason).

I've changed this to:

https://github.com/davwheat/world-cup-of-tocs/blob/8442c37ce58a833abccc6c17d4ebede13275842b/.github/workflows/smoketest.yml#L162

**Comment by Gum-Joe** [comment]
> This could have led to very bad results...
> 
> https://github.com/davwheat/world-cup-of-tocs/blob/60e70c091fd3e08d94630b80a968791b34c11a5c/.github/workflows/build-deploy.yml#L126
> 
> Any push to a branch containing `master` in it would result in a deploy, as well as a PR from `master` to another branch (if we ever wanted to do that for some mad reason).
> 
> I've changed this to:
> 
> https://github.com/davwheat/world-cup-of-tocs/blob/8442c37ce58a833abccc6c17d4ebede13275842b/.github/workflows/smoketest.yml#L162

Yes, good point and something I've missed, that all looks good!
CI script now looks much better.

File: .github/workflows/smoketest.yml [hunk 1]
```diff
@@ -0,0 +1,161 @@
+name: Smoketest
```
Comment (Gum-Joe) [inline .github/workflows/smoketest.yml:1]: Any reasons for the name Smoketest?

File: package.json [hunk 2]
```diff
@@ -40,10 +40,19 @@
     "uuid": "^8.3.2"
   },
   "devDependencies": {
+    "@cherihung/eslint-config-gatsby": "^1.0.7",
```
Comment (Gum-Joe) [inline package.json:43]: Whoops! I should have remkved this earlier...

**Review by Gum-Joe — COMMENTED** [review]
Was just seeing how the review button in GitHub mobile works (thankfully, you can't approve your own PR!)

File: .github/workflows/smoketest.yml (hunk 1 above)
Comment (davwheat) [reply .github/workflows/smoketest.yml:1]: I use this on other projects, and I just think it's a mildly amusing but accurate name: we don't want the project to go up in smoke!

Comment (Gum-Joe) [reply .github/workflows/smoketest.yml:1]: xD that's actually pretty good!

**Review by josh-justjosh — COMMENTED** [review]
Could the problem be with the image files themselves? [social.png](https://github.com/davwheat/world-cup-of-tocs/blob/add-ci/static/social.png) and [icon.png](https://github.com/davwheat/world-cup-of-tocs/blob/add-ci/src/assets/icon.png) don't seem to be showing correctly on GitHub (compared with [icon.png](https://github.com/davwheat/world-cup-of-tocs/blob/master/icon.png) in the main branch)

**Comment by Gum-Joe** [comment]
> Could the problem be with the image files themselves? [social.png](https://github.com/davwheat/world-cup-of-tocs/blob/add-ci/static/social.png) and [icon.png](https://github.com/davwheat/world-cup-of-tocs/blob/add-ci/src/assets/icon.png) don't seem to be showing correctly on GitHub (compared with [icon.png](https://github.com/davwheat/world-cup-of-tocs/blob/master/icon.png) in the main branch)

Thank you for this suggestion! Can confirm, replacing the icon fixed it.

**Comment by Gum-Joe** [comment]
Ok so changing out the icon didn't work, but I stilk think that is the issue is the icon as it looks like it has been corrupted when pushed into github.

Will need to try just creating a new file for the icon so that it's commit history is creatrd from scratch.

**Comment by davwheat** [comment]
@Gum-Joe Any review comments before merge? I ended up making a lot of changes on the PR, so I probably shouldn't review it.

**Review by Gum-Joe — COMMENTED** [review]
Nope. all looks good to me!

**Comment by Gum-Joe** [comment]
(would approve but can't.  Just approve it yourself and merge)

**Review by davwheat — APPROVED** [review]

**Pull request merged**